src/switchroot/ostree-remount.c

```diff
@@ -124,25 +106,10 @@ main(int argc, char *argv[])
       exit (EXIT_SUCCESS);
     }
 
-  /* Query the repository configuration - this is an operating system builder
-   * choice.
-   * */
-  const bool sysroot_readonly = sysroot_is_configured_ro ();
-
-  /* Mount the sysroot read-only if we're configured to do so.
-   * Note we only get here if / is already writable.
-   */
-  do_remount ("/sysroot", !sysroot_readonly);
-
-  if (sysroot_readonly)
+  /* Handle remounting /sysroot read-only now */
+  if (unlink (_OSTREE_SYSROOT_READONLY_STAMP) == 0)
     {
-      /* Now, /etc is not normally a bind mount, but remounting the
-       * sysroot above made it read-only since it's on the same filesystem.
-       * Make it a self-bind mount, so we can then mount it read-write.
-       */
-      if (mount ("/etc", "/etc", NULL, MS_BIND, NULL) < 0)
-        err (EXIT_FAILURE, "failed to make /etc a bind mount");
-      do_remount ("/etc", true);
+      do_remount ("/sysroot", false);
```

Comment:
Couldn't we ensure that the sysroot is mounted read-only from `ostree-prepare-root`? That way there's no gap either in the real root as for when `sysroot.readonly` is enforced.

Reply:
That'd make a lot of sense, but I'd prefer that as a followup cleanup because it affects a whole lot more of the code paths, mostly around handling of `/var`.  And thing like the `/run/ostree/initramfs-mount-var` path versus the fstab generator etc.

Reply:
Right yeah. OK filed https://github.com/ostreedev/ostree/issues/2115 as a follow up.